pkg/kv/transport.go

```diff
@@ -136,15 +138,28 @@ func grpcTransportFactoryImpl(
 	// Put known-unhealthy clients last.
 	splitHealthy(clients)
 
+	var ds *DistSender
+	var ok bool
+	if opts.invokedSender != nil {
+		ds, ok = opts.invokedSender.(*DistSender)
+		if !ok {
+			ds = nil
+		}
+	} else {
+		ds = nil
+	}
```

Comment:
I think this could be simplified to:

```
ds, _ := opts.invokedSender.(*DistSender)
```

You need the two return value form so that the type assertion doesn't panic, but you don't actually need the second result as `ds` will be set to `nil`.